lib/jellyfish_web/server_socket.ex

```diff
@@ -73,10 +81,32 @@ defmodule JellyfishWeb.ServerSocket do
     end
   end
 
+  def handle_in({encoded_message, [opcode: :binary]}, state) do
+    with %ServerMessage{content: request} <- ServerMessage.decode(encoded_message),
+         {:room_state_request, %RoomStateRequest{content: {_variant, option}}} <- request do
+      room_state = get_room_state(option)
+
+      reply =
+        %ServerMessage{content: room_state}
+        |> ServerMessage.encode()
+
+      {:reply, :ok, {:binary, reply}, state}
+    else
+      _other ->
```

Comment:
In case of failure in the second clause, you'd probably want to inspect the unmatched value (`_other`), not the encoded message